pkg/network/server.go

```diff
@@ -402,6 +414,19 @@ func (s *Server) handleBlockCmd(p Peer, block *core.Block) error {
 	return s.bQueue.putBlock(block)
 }
 
+// handlePing processes ping request.
+func (s *Server) handlePing(p Peer, ping *payload.Ping) error {
+	return p.WriteMsg(NewMessage(s.Net, CMDPong, payload.NewPing(s.id, s.chain.BlockHeight())))
+}
+
+// handlePing processes pong request.
+func (s *Server) handlePong(p Peer, pong *payload.Ping) error {
```

Comment:
I'd check here that we really do expect a pong (that is, we'd sent a ping some time ago and it wasn't replied to yet).

Reply:
added separate PingTimeout which currently equals to PingInterval (didn't want to mess up with two of them). But anyway don't see any problem with having one constant for that

Reply:
It not the same, I'm talking about spurious pongs received from the other side which technically violate the protocol. That's what I want to be able to detect and probably log something about it (or even drop the peer).